feat(ucs): add new resource policy (#3126)
huaweicloud/terraform-provider-huaweicloud: docs/resources/ucs_policy.md

---
subcategory: "Ubiquitous Cloud Native Service (UCS)"
---

# huaweicloud_ucs_policy

Manages a UCS policy resource within HuaweiCloud.

## Example Usage

### Basic Usage

```hcl

> variable "user_id_1" {}
> variable "user_id_1" {}

Reviewer: ```suggestion
variable "user_id_1" {}
variable "user_id_2" {}
```
or
```
variable "user_ids" {
  type = list(string)
}
```

Author: fixed

resource "huaweicloud_ucs_policy" "test" {
  name         = "policy-1"
  iam_user_ids = [var.user_id_1, var.user_id_2]
  type         = "admin"
  description  = "created by terraform"
}
```

### Custom Type Policy

```hcl
variable "user_id_1" {}

> variable "user_id_1" {}

Reviewer: ditto

Author: fixed

resource "huaweicloud_ucs_policy" "test" {
  name         = "policy-1"
  iam_user_ids = [var.user_id_1, var.user_id_2]
  type         = "custom"
  description  = "created by terraform"

  contents {
    verbs     = ["*"]
    resources = ["*"]
  }
}
```

## Argument Reference

The following arguments are supported:

* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.

* `name` - (Required, String, ForceNew) Specifies the name of the UCS policy. The name consists of 1 to 63 characters,
  including lowercase letters, digits and hyphens (-), must start with a letter and end with a letter or digit.

  Changing this parameter will create a new resource.

> * `iam_user_ids` - (Required, List) Specifies the list of iam user IDs to associate to the policy.

Reviewer: ```suggestion
* `iam_user_ids` - (Required, List) Specifies the list of IAM user IDs to associate to the policy.
```

Author: fixed

* `type` - (Required, String) Specifies the type of the UCS policy.
  The value can be: **readonly**, **develop**, **admin** and **custom**.

* `description` - (Optional, String) Specifies the description of the UCS policy.
  The description consists of 0 to 255 characters.

* `details` - (Optional, List) Specifies the details of the UCS policy.
  This only works when the type is **custom**.
  The [Details](#Policy_Details) structure is documented below.

<a name="Policy_Details"></a>

> The `Content` block supports:

Reviewer: ```suggestion
The `details` block supports:
```

Author: fixed

* `operations` - (Optional, List) Specifies the list of operations.

* `resources` - (Optional, List) Specifies the list of resources.

## Attributes Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

* `created_at` - The created time.

* `updated_at` - The updated time.

## Import

> The health check can be imported using the `id`, e.g.

Reviewer 2: ```suggestion
The UCS policy can be imported using the `id`, e.g.
```

Author: fixed
